core/app/models/spree/order.rb

```diff
@@ -400,7 +400,7 @@ def deliver_order_confirmation_email
 
     # Helper methods for checkout steps
     def paid?
-      payment_state == 'paid' || payment_state == 'credit_owed'
+      %w(paid credit_owed).include?(payment_state)
```

Comment:
The problem is all that "seems" in the argument.

You have too options:

* Pull the Spree stuff you need, but nothing else. Than fixes.
* Fix all stuff you can and intermix with Spree stuff.

But what happens here seems arbitrary.

I do not see the reason why all the other, non spree cherry-pick commits are okay, but mine would not be.